inst/htmlwidgets/bpmnVisualizationR.js

```diff
@@ -50,23 +50,29 @@ HTMLWidgets.widget({
                 }
             };
         }
+        
+        function buildDefaultOverlayPosition(isShape) {
+            return isShape ? 'top-center' : 'middle';
+        }
 
         return {
-            renderValue: function(x) {
-                bpmnVisualization.load(x.bpmnContent, { fit: {type: bpmnvisu.FitType.Center, margin: 30} });
-                // Add overlays
-                x.overlays?.map(overlay => {
-                    const elementsByIds = bpmnVisualization.bpmnElementsRegistry.getElementsByIds(overlay.elementId);
-
-                    if (elementsByIds.length) {
-                        const overlayConfig = {
-                            label: overlay.label,
-                            style: overlay.style ?? (x.enableDefaultOverlayStyle && buildDefaultOverlayStyle(elementsByIds[0].bpmnSemantic.isShape)),
-                            position: elementsByIds[0].bpmnSemantic.isShape ? 'top-center' : 'middle',
-                        };
+            renderValue: function({bpmnContent, overlays, enableDefaultOverlayStyle}) {
```

Comment:
**praise:** nice to use object destructuring. This makes the code easier to read!